jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2524,7 +2524,90 @@ public boolean startsWith(String prefix) {
 	public boolean startsWith(String prefix, int start) {
 		return regionMatches(start, prefix, 0, prefix.lengthInternal());
 	}
-
+	
+/*[IF Java11]*/
+	/**
+	 * Strip leading and trailing white space from a string.
+	 * 
+	 * @return a substring of the original containing no leading
+	 * or trailing white space
+	 * 
+	 * @since 11
+	 */
+	public String strip() {
+		String result;
+		
+		if (enableCompression && (null == compressionFlag || coder == LATIN1)) {
+			result = StringLatin1.strip(value);
+		} else {
+			result = StringUTF16.strip(value);
+		}
+		
+		if (result == null) {
+			if (0 == lengthInternal()) {
```

Comment:
As far as I can tell the StringLatin1 and StringUTF16 `strip` methods already check for this, so there is no need for a double check. Similarly for other methods.

Reply:
good point. fixed that and retested.